solnlib/splunk_rest_client.py

```diff
@@ -98,10 +100,18 @@ def _request_handler(context):
     else:
         cert = None
 
+    retries = Retry(
+        total=MAX_REQUEST_RETRIES,
+        backoff_factor=0.3,
+        status_forcelist=[500, 502, 503, 504],
+        allowed_methods=["GET", "POST", "PUT", "DELETE"],
+        raise_on_status=False,
+    )
```

Comment:
Should get config from context and fill with defaults when not present

Reply:
You're right, that's the plan. We're going to refactor this code soon.

Reply:
@dkvashninsplunk this was done as a temporary solution to unblock other team

we want to get rid of `requests` library at all for `solnlib` and `splunktaucclib` so this code might potentially be deleted once we have a solution